experimental/traffic-portal/src/app/api/testing/server.service.ts

```diff
@@ -302,4 +317,55 @@ export class ServerService {
 		srv.statusId = status.id;
 		srv.offlineReason = offlineReason ?? null;
 	}
+
+	/**
+	 * Creates a status.
+	 *
+	 * @param status The status details (name & description) to create.
+	 * @returns The status as created and returned by the API.
+	 */
+	public async createStatus(status: RequestStatus): Promise<ResponseStatus> {
+		const newStatus = {
+			...status,
+			id: ++this.statusIdCounter,
+			lastUpdated: new Date()
+		} as { description: string; id: number; lastUpdated: Date; name: string };
+		this.statuses.push(newStatus);
+		return newStatus;
+	}
+
+	/**
+	 * Updates status Details.
+	 *
+	 * @param payload containes name and description for the status., unique identifier thereof.
+	 * @param id The Status ID
+	 */
+	public async updateStatusDetail(payload: ResponseStatus, id: number): Promise<ResponseStatus> {
+		const index = this.statuses.findIndex(u => u.id === id);
+		if (index < 0) {
+			throw new Error(`no such status with id: ${id}`);
+		}
+		const updated = {
+			...payload,
+			lastUpdated: new Date()
+		} as { description: string; id: number; lastUpdated: Date; name: string };
+		this.statuses[index] = updated;
+
+		return updated;
+	}
+
+	/**
+	 * Deletes a Status.
+	 *
+	 * @param id The ID of the Status to delete.
+	 * @returns The response.
+	 */
+	public async deleteStatus(id: number): Promise<HttpResponse<object> | null> {
```

Comment:
This function's call signature is not assignable to the call signature of the concrete implementation; `HttpResponse<object> | null` is not assignable to `ResponseStatus`. In order for the testing services to work as drop-in replacements for the "real" services, they need to return the same kind of data.